src/tokenizer.cpp

```diff
@@ -95,10 +95,7 @@ tokenizer_t::tokenizer_t(const wchar_t *start, tok_flags_t flags)
 
 tok_t::tok_t(token_type_t type) : type(type) {}
 
-/// Tests if this character can be a part of a string. The redirect ^ is allowed unless it's the
-/// first character. Hash (#) starts a comment if it's the first character in a token; otherwise it
-/// is considered a string character. See issue #953.
-static bool tok_is_string_character(wchar_t c, bool is_first) {
+bool tok_is_string_character(wchar_t c, bool is_first, maybe_t<wchar_t> next) {
```

Comment:
Hmm here I was probably wrong to use "word character", because chars like `"$/` are included here.

The fish lexer knows 8 different token kinds (see `token_type_t`); and basically everything that is not a separator is a "string".

I think I'll just change word -> token in the user-facing parts.